engine/chunk/src/main/java/io/deephaven/chunk/util/pools/MultiChunkPool.java

```diff
@@ -29,37 +62,37 @@ public static MultiChunkPool forThisThread() {
         return POOL_THREAD_LOCAL.get();
     }
 
-    private final BooleanChunkPool booleanChunkPool = new BooleanChunkPool();
-    private final CharChunkPool charChunkPool = new CharChunkPool();
-    private final ByteChunkPool byteChunkPool = new ByteChunkPool();
-    private final ShortChunkPool shortChunkPool = new ShortChunkPool();
-    private final IntChunkPool intChunkPool = new IntChunkPool();
-    private final LongChunkPool longChunkPool = new LongChunkPool();
-    private final FloatChunkPool floatChunkPool = new FloatChunkPool();
-    private final DoubleChunkPool doubleChunkPool = new DoubleChunkPool();
-    private final ObjectChunkPool objectChunkPool = new ObjectChunkPool();
+    private final BooleanChunkPool booleanChunkPool = new BooleanChunkSoftPool();
+    private final CharChunkPool charChunkPool = new CharChunkSoftPool();
+    private final ByteChunkPool byteChunkPool = new ByteChunkSoftPool();
+    private final ShortChunkPool shortChunkPool = new ShortChunkSoftPool();
+    private final IntChunkPool intChunkPool = new IntChunkSoftPool();
+    private final LongChunkPool longChunkPool = new LongChunkSoftPool();
+    private final FloatChunkPool floatChunkPool = new FloatChunkSoftPool();
+    private final DoubleChunkPool doubleChunkPool = new DoubleChunkSoftPool();
+    private final ObjectChunkPool objectChunkPool = new ObjectChunkSoftPool();
 
-    private final Map<ChunkType, ChunkPool> pools;
+    private final Map<ChunkType, Supplier<ChunkPool>> pools;
```

Comment:
I know it feels silly since the generic `ChunkPool` interface is unused, but it seems to me like we could use the `asChunkPool` methods once each to populate the map, instead of invoking the supplier on demand. It would probably use no more space, etc.

Reply:
Good call - I had it that way to begin with, but rewrote it this way, easy to revert.